src/module/data/pseudo-documents/power-roll-effects/base-power-roll-effect.mjs

```diff
@@ -155,6 +154,14 @@ export default class BasePowerRollEffect extends TypedPseudoDocument {
 
   /* -------------------------------------------------- */
 
+  /**
+   * Add listeners to elements in the pseudo document sheet specific to this subtype
+   * @param {HTMLElement} html
+   */
+  async onRender(html) {}
```

Comment:
Overall I'm not a big fan of this design paradigm. I would look in the sheet for the sheet logic and in the data models for their data model logic. I completely understand that it makes *sense* that the model subtype handles model-specific sheet rendering, but perhaps we'd be happier in the long run subclassing the sheet?